packages/core/src/stateUtils.ts

```diff
@@ -1550,6 +1554,7 @@ function resolveAndExecuteActionsWithContext(
           : undefined;
 
     function executeAction() {
+      executingCustomAction = resolvedAction;
```

Comment:
i'd move this into the try block, it will be more clear that it goes together with the reset in the finally block